README.md

```diff
@@ -0,0 +1,155 @@
+vagrant$ lspci
+00:08.0 Ethernet controller: Red Hat, Inc Virtio network device
+00:09.0 Ethernet controller: Red Hat, Inc Virtio network device
+```
+
+To use `Capsule`, add it as a dependency in your `Cargo.toml`,
+
+```
+[dependencies]
+capsule = "0.1"
+```
+
+You should also mount the working directory of your project as a volume for the sandbox. Then you can use `cargo` commands inside the container as normal.
+
+### Linux Distributions
+
+Alternatively, if you are already running a Linux operating system and do not wish to use `Vagrant`, you should be able to run the sandbox container directly. We've tested the sandbox on `Debian Buster`, `Ubuntu Bionic` and `CentOS 7`. Other Linux distributions and versions may work similarly with minor tweaks, if you are not running the versions we tested on.
+
+You need to make a few configuration changes to support `DPDK`.
+
+`DPDK` needs a small kernel module to set up the device, map device memory to user-space and register interrupts. The standard `uio_pci_generic` module included in the Linux kernel can provide the capability. To load the module, use the command,
+
+```
+host$ sudo modprobe uio_pci_generic
+```
+
+Debian and CentOS by default include this module. If you are on an Ubuntu host, `uio_pci_generic` is not part of the base system. You will need to install an additional package,
+
+```
+sudo apt install linux-modules-extra-$(uname -r)
+```
+
+DPDK also needs `HugePages` support for the large memory pool allocation used for packet buffers. To enable `HugePages`, use the following commands,
+
+```
+host$ sudo su
+host$ echo "vm.nr_hugepages = 2048" >> /etc/sysctl.conf
+host$ sysctl -e -p
+host$ exit
+```
+
+Before your `Capsule` application can access a network interface on the host, the interface must be bound to a DPDK compatible driver with the [`dpdk-devbind`](https://doc.dpdk.org/guides/tools/devbind.html) utility. To bind the driver, find the interface's PCI address and use the command,
+
+```
+host$ docker pull getcapsule/dpdk-devbind:18.11.6
+host$ docker run --rm --privileged --network=host \
+    -v /lib/modules:/lib/modules \
+    getcapsule/dpdk-devbind:18.11.6 \
+    /bin/bash -c 'dpdk-devbind.py --force -b uio_pci_generic #PCI_ADDR#'
+```
+
+Once you've made the necessary changes, you can pull down the sandbox container and run it,
+
+```
+host$ docker pull getcapsule/sandbox:18.11.6-1.42
+host$ docker run ...
+```
+
+### Without Docker
+
+If you choose not to use `Docker`, or if you are using a Linux distribution incompatible with the `Debian` based sandbox, then you need to [install DPDK from source](https://doc.dpdk.org/guides/linux_gsg/build_dpdk.html) yourself.
+
+In additional to the [required tools and libraries](https://doc.dpdk.org/guides/linux_gsg/sys_reqs.html#compilation-of-the-dpdk), you also need the library for packet captures to run some `Capsule` example applications. Install `libpcap-dev` for `Debian`/`Ubuntu` or `libpcap-devel` for `RHEL`/`CentOS`/`Fedora`.
```

Comment:
sp. In addition

Reply:
"you also need the `pcap` library" ... ?

Reply:
people might not know what pcap is?

Reply:
Maybe packet captures (pcaps)